tests/components/wallbox/test_init.py

```diff
@@ -1,27 +1,30 @@
 """Test Wallbox Init Component."""
-import json
+from http import HTTPStatus
+from unittest.mock import Mock, patch
 
-import requests_mock
+from requests.exceptions import HTTPError
 
-from homeassistant.components.wallbox import CHARGER_MAX_CHARGING_CURRENT_KEY
-from homeassistant.config_entries import ConfigEntryState
+from homeassistant.config_entries import ConfigEntryState, UnknownEntry
 from homeassistant.core import HomeAssistant
 
-from . import test_response
-
-from tests.components.wallbox import (
+from . import (
     DOMAIN,
-    authorisation_response,
     entry,
     setup_integration,
+    setup_integration_charger_status_connection_error,
     setup_integration_connection_error,
-    setup_integration_read_only,
+    setup_integration_invalidauth_error,
 )
 
 
 async def test_wallbox_setup_unload_entry(hass: HomeAssistant) -> None:
     """Test Wallbox Unload."""
 
+    try:
```

Comment:
You can use `MockConfigEntry` here

Reply:
I'm using a MockConfigEntry, I'm importing it from __init__. I'm unloading in every test to ensure that they run (even when another test fails and skips the unloading part) also, I'm loading in every test because I sometimes trigger an error during loading for test purposes. 
Alternatively I could test if the entry is loaded at the start of the test and only unload / load when necessary for testing.

Reply:
Even if it fails to unload in a previous test it shouldn't matter since each test should get a fresh test Home Assistant instance.

Reply:
I gave it another go, but if I don't load / unload in every test, the tests fail. Maybe that's because I centralized the `setup_integration` function in __init__?